06-building-a-program-start-to-finish/solving-the-problem/output/invertedindex.js

```diff
@@ -0,0 +1,222 @@
+
+const hasProperty = Object.prototype.hasOwnProperty;
+let instance = null;
+
+class InvertedIndex {
+  /**
+    * constructor method
+  */
+  constructor() {
+    if (!instance) {
+      instance = this;
+      this.filesIndexed = {};
+      this.inputData = {};
+      this.error = {};
+    }
+    return instance;
+  }
+
+  /**
+   * readFile function is used to get all the index
```

Comment:
Only validates if file is correct json file and return JSON.parse if true and false if otherwise